### DISCUSSION TITLE
Not rendering correctly, using Nerd Fonts

### DISCUSSION THREAD

**Discussion by JayG30** [JanDeDobbeleer/oh-my-posh#5855] [Troubleshoot] [answered]
I followed the installation docs.
1. Install Windows Terminal
2. Install via winget (winget install JanDeDobbeleer.OhMyPosh -s winget)
3. Install a font (oh-my-posh font install meslo)
4. Set Default Profile -> Appearance -> Font Face to Nerd Font (MesloLGM Nerd Font)
5. Set default theme to test (oh-my-posh init pwsh --config "$env:POSH_THEMES_PATH/jandedobbeleer.omp.json" | Invoke-Expression)

Yet this is what I see when I launch a Terminal without Administrator permissions;
![image](https://github.com/user-attachments/assets/5dc6b221-8157-436e-9ba8-361ba0290517)

When I launch Terminal as Administrator then it works;
![image](https://github.com/user-attachments/assets/1aa4cb1d-a11a-4790-9fc5-7db242131ff4)

Everything is updated and installed within the past hour.
I've also tried a different Nerd Font (Fira Code), didn't change behavior.
Looked at the other themes, same behavior.
Checked that the fonts exist in Windows Fonts control panel, they do.

I don't have a clue what is causing this and I can't find anything online that helps. Everyone that has a problem like this seems to always come back to they didn't install a Nerd Font or set it. But I have.
![image](https://github.com/user-attachments/assets/9246463b-790f-49f0-895e-41b07ca027d8)

**Comment by JanDeDobbeleer** [comment]
@JayG30 try forcing PowerShell to UTF8. This could be the issue.

**Comment by JayG30** [comment, marked as answer]
I figured out the issue. Powershell was launching in ConstrainedLanguage mode.
I will leave this here for others.

I've used this PC over the years to create and test system application control policies through AppLocker, SRP, and Windows Defender Application Control (WDAC) for my job. And while it at least appeared I had removed and wasn't enforcing anything, there were still things lingering on the system which by design Powershell automatically detects and puts you into ConstrainedLanguage mode. There is no way to switch to FullLanguage mode without removing or creating rules to allow it.

This website [HERE](https://www.deploymentresearch.com/psscriptpolicytest-script-gets-blocked-by-applocker-in-the-event-log-why-and-what-are-those-files/) provided some clarity that AppLocker was the culprit. When you launch powershell it creates a script in $env:temp and tries to run it. If it fails it puts you into ConstrainedLanguage mode. More importantly you will see in Event Viewer an ERROR from AppLocker.

I had already done this, but you can remove AppLocker rules by using local group policy, navigate to AppLocker, right click and clear policy. This will remove all rules and set everything to disabled.

What went wrong for me was that I did this while the window service "Application Identity" was not running (disabled by default). This service is requires to create AND delete AppLocker rules. This is explained in the Microsoft documentation [HERE](https://learn.microsoft.com/en-us/windows/security/application-security/application-control/app-control-for-business/applocker/delete-an-applocker-rule). If you clear the policy through GPO while this is NOT enabled you have to go into C:\Windows\System32\AppLocker and delete the files.

The reason this works when you "run as Administrator" is because the AppLocker policies are typically setup by default to exclude this built-in account.

Now that I'm not stuck in ConstrainedLanguage mode it works.
